federated_aws_cli/cache.py

```diff
@@ -107,22 +155,24 @@ def read_aws_shared_credentials():
 
 
 @_requires_safe_cache_dir
-def write_aws_shared_credentials(credentials):
-    # Create a sha256 of the role arn, so fix length and remove weird chars
+def write_aws_shared_credentials(credentials, role_arn, role_map=None):
     path = os.path.join(DOT_DIR, "credentials")
 
     # Try to read in the existing credentials
     config = read_aws_shared_credentials()
 
-    # Add all the new credentials to the config object
-    for section in credentials.keys():
-        if not config.has_section(section):
-            config.add_section(section)
-
-        logger.debug("The section is: {}".format(credentials[section]))
+    # such as infosec-somerole
+    profile = _role_to_profile_name(role_arn, role_map)
 
-        for key, value in viewitems(credentials[section]):
-            config.set(section, key, value)
+    # Add all the new credentials to the config object
+    if not config.has_section(profile):
+        config.add_section(profile)
+        logger.debug("Added new profile: {}".format(profile))
+
+    # Update the values
+    for cred_key, aws_key in viewitems(CREDENTIALS_TO_AWS_MAP):
+        if cred_key in credentials:
+            config.set(profile, aws_key, credentials[cred_key])
 
     # Order all the sections alphabetically
```

Comment:
We should probably retain the order that the config is in (in case the user has added comments or has a preference). Is it possible to retain the order?

Reply:
I just tested and this doesn't change the order of anything but the section names.

Reply:
Ok cool. Does it retain comments or do those get lost?

Reply:
After doing some research, I've found a way to preserve comments (and their order) with Python 3. There is no such way to do so with Python 2 unfortunately. I'll add that to the PR.